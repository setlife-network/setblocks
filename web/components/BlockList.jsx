import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';
import { cloneDeep } from 'lodash';

import SetBlock from './SetBlock';
import Text from './Text';
import { DEFAULT_SETBLOCKS } from '../constants';

import {
    setEnableSubmit, updateUnsavedSetblocks,
} from '../reducers/environment';

class BlockList extends React.Component {

    completeWithEmptySetBlocks(setBlocks) {
        const defaultSetBlocks = cloneDeep(DEFAULT_SETBLOCKS);
        let replacedBlocks = 0;
        setBlocks = _.orderBy(setBlocks, ['blockTime'], ['asc']); // Use Lodash to sort array by 'name'
        setBlocks = _.uniqBy(setBlocks, 'blockTime'); // Use Lodash to delete repeated blockTimes
        if (setBlocks && setBlocks.length > 0) { // To avoid iterate if the array is empty, return default
            defaultSetBlocks.forEach((setBlock, index, theArray) => {
                if (setBlocks && replacedBlocks < setBlocks.length && setBlock.blockTime === setBlocks[replacedBlocks].blockTime) {
                    theArray[index] = setBlocks[replacedBlocks];
                    replacedBlocks++;
                }
            });
        }
        return defaultSetBlocks
    }

    updateUnsavedSetBlock(selectedDay, index, editedSetBlock ) {
        const { unsavedSetBlocks } = this.props
        let dayEdited = unsavedSetBlocks[selectedDay];
        dayEdited[index] = editedSetBlock; // index 0 = SetBlock with blockTime 1
        this.props.updateUnsavedSetblocks({
            ...unsavedSetBlocks,
            [selectedDay]: dayEdited
        })
        this.props.setEnableSubmit(true)
    }

    render() {
        const {
            editModeSchedule, currentWeeklySetblocks, unsavedSetBlocks, selectedDay 
        } = this.props;
        const selectedDayFormatted = moment(selectedDay).format('YYYY-MM-DD')
        let setBlocksByDate =  _.groupBy(currentWeeklySetblocks, 'date');
        let setBlocks = setBlocksByDate[selectedDayFormatted];

        if (editModeSchedule && unsavedSetBlocks ) {
            // As it is your schedule, you can see the empty blocks, to complete then, that's why it is completed with the missing ones
            return this.completeWithEmptySetBlocks(setBlocks).map((setBlock, index) => {
                return (
                    <SetBlock
                        data={setBlock}
                        key={setBlock.id || (index + selectedDayFormatted)}
                        editMode={editModeSchedule}
                        updateSetBlock={(editedSetBlock) => this.updateUnsavedSetBlock(selectedDayFormatted, index, editedSetBlock )}
                    />
                )
            })
        } else if (!editModeSchedule && setBlocks) {
            setBlocks = _.orderBy(setBlocks, 'blockTime') // To properly render in order
            return setBlocks.map((setBlock, index) => {
                return <SetBlock data={setBlock} key={setBlock.id || index} />
            })
        } else {
            return (
                <Text weight='600' align='center'> This user hasn't committed any Setblocks for this day </Text>
            )
        }
    }
}

const mapStateToProps = ({ environment }) => {
    return {
        ...environment
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        setEnableSubmit: (enableSubmit) => dispatch(setEnableSubmit(enableSubmit)),
        updateUnsavedSetblocks: (unsavedSetBlocks) => dispatch(updateUnsavedSetblocks(unsavedSetBlocks))
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(BlockList);