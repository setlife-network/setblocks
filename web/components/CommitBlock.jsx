import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { Check } from 'styled-icons/feather/Check.cjs'

import Card from './Card';
import Flex from './Flex';
import Text from './Text';

import {
    createSetBlock, deleteSetblock, setEditModeSchedule, updateSetBlock 
} from '../reducers/environment';

class CommitBlock extends React.Component {

    state = {
        showToast: false
    }

    upsertSetBlocks = () => {
        const { selectedDay, unsavedSetBlocks, currentTeamMember, enableSubmit } = this.props
        const day = moment(selectedDay).format('YYYY-MM-DD');
        if (enableSubmit) {

            unsavedSetBlocks[day].map( (setBlock) => {
                if (setBlock.id) {
                // Update
                    if (setBlock.blockFraction !== 0) {
                    // Update if change description, issueUrl, or blockFraction
                        this.props.updateSetBlock(setBlock)
                    } else {
                    // Delete if blockFraction is 0
                        this.props.deleteSetblock({ setblockId: setBlock.id })
                    }
                } else if (setBlock.blockFraction !== 0) {
                // Create a new one if have blockFraction != 0
                    this.props.createSetBlock({
                        teamMemberId: currentTeamMember.id,
                        date: day,
                        ...setBlock,
                        issueUrl: (setBlock.issueUrl || '')
                    })
                }
            })

            this.setState({
                showToast: true // Improve toast system in a future
            })
            _.delay( () => { // This probably made a warning but is only temporal, this will be removed with the toast system
                this.setState({
                    showToast: false // Improve toast system in a future
                })
                this.props.setEditModeSchedule(false);
            }, 1000);
        }
    }

    countHours() {
        const { selectedDay, unsavedSetBlocks } = this.props
        const day = moment(selectedDay).format('YYYY-MM-DD');
        return unsavedSetBlocks[day]
            ? unsavedSetBlocks[day].reduce( (last, setBlock) => {
            // Math.abs is used because the second blockFraction is represented by -0.5
                return last + (Math.abs(setBlock.blockFraction) * 4)
            }, 0)
            : 0 // If not defined yet, return 0
    }

    countBlockFractions() {
        const { selectedDay, unsavedSetBlocks } = this.props
        const day = moment(selectedDay).format('YYYY-MM-DD');
        return unsavedSetBlocks[day]
            ? unsavedSetBlocks[day].reduce( (last, setBlock) => {
            // Math.abs is used because the second blockFraction is represented by -0.5
                return last + (Math.abs(setBlock.blockFraction))
            }, 0)
            : 0 // If not defined yet, return 0
    }

    render() {
        const { selectedDay, enableSubmit } = this.props
        const { showToast } = this.state
        const onSaveToast = (
            <Text color='accent'>
                {'SetBlock committed successfully!!'}
            </Text>
        )

        return (
            <Flex column center width='100%'>
                <Text
                    weight='600'
                    aling='center'
                    size='10px'
                    ml='-10rem'
                >
                    {this.countHours() + ' work hours scheduled'}
                </Text>

                <Flex row center>
                    {enableSubmit
                    && (
                        <Text
                            weight='600'
                            aling='center'
                            size='10px'
                            whiteSpace='pre'
                            mx='0.5rem'
                        >
                            {' Commit ' + this.countBlockFractions() + ' SetBlocks \n '
                        + 'for ' + moment(selectedDay).format('dddd, MMM Do') + '?'}
                        </Text>
                    )}
                    <Card
                        bg={enableSubmit ? '#F5F5F6' : 'overlay'}
                        minWidth='20%'
                        maxWidth='100px'
                        height='40px'
                        mx='auto'
                        borderRadius='5px'
                        depth={7}
                        onClick={() => this.upsertSetBlocks()}
                    >
                        <Text color='textSecondary' align='center' size='1rem' weight='600' mt='8px' mx='5px'>
                            {'Commit'}
                            {React.createElement(Check, {
                                size: 24,
                                color: 'black'
                            })}
                        </Text>
                    </Card>
                </Flex>
                {showToast && (onSaveToast)}
            </Flex>
        );
    }
}

const mapStateToProps = ({ environment }) => {
    return {
        ...environment
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        createSetBlock: (params) => dispatch(createSetBlock(params)),
        deleteSetblock: (params) => dispatch(deleteSetblock(params)),
        updateSetBlock: (params) => dispatch(updateSetBlock(params)),
        setEditModeSchedule: (editMode) => dispatch(setEditModeSchedule(editMode))
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CommitBlock);
