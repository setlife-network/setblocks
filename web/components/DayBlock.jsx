import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';

import Card from './Card';
import Flex from './Flex';
import Text from './Text';

import theme from '../styles/theme'

class DayBlock extends React.Component {

    renderTinySetBlocks = (setBlocks, day, fetchingData) => {
        const blockDay = moment(day).format('YYYY-MM-DD')
        const setBlocksByDate = _.groupBy(setBlocks, 'date')
        let setBlocksToRender = setBlocksByDate[blockDay] || [];
        setBlocksToRender = _.orderBy(setBlocksToRender, 'blockTime') // Order by blockTime to have all the tinyBlock ordered
        if (setBlocksToRender.length > 0 && !fetchingData) {
            return setBlocksToRender.map(setBlock => (
                <Card
                    key={setBlock.id}
                    height='8px'
                    width='8px'
                    borderBottom={setBlock.blockFraction === 1.0 ? `4px ${theme.colors.accent} solid` : (setBlock.blockFraction === 0.5 ? `4px ${theme.colors.accent} solid` : '')}
                    borderTop={setBlock.blockFraction === 1.0 ? `4px ${theme.colors.accent} solid` : (setBlock.blockFraction === -0.5 ? `4px ${theme.colors.accent} solid` : '')}
                    bg='primaryLight'
                    my='0.3rem'
                    mr='0.3rem'
                >
                </Card>
            ))
        } else {
            return <Card height='8px' width='5px' bg='none' my='0.3rem' mr='0.3rem' /> // To prevent the sidebar grows slightly in width when render the tinySetBlocks
        }
    }

    render() {
        const { day, selected, onClick, fetchingData, currentWeeklySetblocks } = this.props

        return (
            <Flex
                column
                center
                className='DayBlock'
                style={selected ? { borderBottom: `2px ${theme.colors.grey} solid`, borderTop: `2px ${theme.colors.grey} solid` } : {}}
            >
                <Card
                    width='100%'
                    bg={selected ? theme.colors.lightGrey : theme.colors.primary}
                    mx='0.5rem'
                    onClick={() => onClick(day)}
                >
                    <Flex row center>
                        <Flex column mx='0.5rem'>
                            { // If you are waiting for the API to respond, it does not render
                                this.renderTinySetBlocks(currentWeeklySetblocks, day, fetchingData)
                            }
                        </Flex>
                        <Flex column mr='auto'>
                            <Text align='center' mb='0rem' size={25} lineHeight='1' color={selected ? 'accent' : theme.colors.overlay}>
                                {day.getDate()}
                            </Text>
                            <Text align='center' mt='0rem' size={11} lineHeight='1' color={selected ? 'accent' : theme.colors.overlay}>
                                {day.toDateString().slice(0, 3)}
                            </Text>
                        </Flex>
                    </Flex>
                </Card>
            </Flex>
        );
    }
}

const mapStateToProps = ({ environment }) => {
    return {
        ...environment
    };
};

const mapDispatchToProps = {
};
export default connect(mapStateToProps, mapDispatchToProps)(DayBlock)