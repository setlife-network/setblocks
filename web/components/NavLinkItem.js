import React from 'react'
import { Link } from 'react-router-dom'

import Box from './Box'
import Flex from './Flex'
import Text from './Text'
import Card from './Card';

import theme from '../styles/theme'

export default ({
    text, selected, icon, route, width
}) => (
    <Box
        width={width + '%'}
    >
        <Link
            to={route}
            style={{ textDecoration: 'none' }}
        >
            <Card
                bg={theme.colors.charcoal}
                width='100%'
                height='100%'
                borderTop={selected ? `2px ${theme.colors.accent} solid` : ''}
                depth={8}
            >
                <Flex
                    column
                    height='100%'
                    py='5px'
                    alignItems='center'
                >
                    {React.createElement(icon, {
                        size: 24,
                        color: theme.colors.white
                    })}
                    <Text
                        color={theme.colors.white}
                        size={[10]}
                        my='0'
                        weight='600'
                    >
                        {text}
                    </Text>
                </Flex>
            </Card>
        </Link>
    </Box>
)
