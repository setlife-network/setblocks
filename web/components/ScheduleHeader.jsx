import React from 'react';
import moment from 'moment';

import Text from './Text';
import Card from './Card';
import Flex from './Flex';
import theme from '../styles/theme';


export default class ScheduleHeader extends React.Component {
    render() {
        const { selectedDay } = this.props;
        return (
            <Flex
                row
                className='ScheduleHeader'
                center
                width='100%'
            >
                <Card
                    width='10px'
                    height='10px'
                    bg={theme.colors.transparent /* For the moment */}
                    borderRadius={50}
                    my='auto'
                    mx='0.5rem'
                />
                <Text
                    size={27}
                    align='center'
                    color={theme.colors.textPrimary}
                    my='0.5rem'
                    mx='auto'
                >
                    {moment(selectedDay).format('dddd Do')}
                </Text>

            </Flex>
        );
    }
}
