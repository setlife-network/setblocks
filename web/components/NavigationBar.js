import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash'

import NavLinkItem from './NavLinkItem';

import { NAVIGATION } from '../constants';
import Flex from './Flex';

export class NavigationBar extends React.Component {
    renderNavItem = (item) => {
        const { path } = this.props;
        const route = '/' + _.split(path, '/', 2)[1];
        return (
            <NavLinkItem
                {...item}
                selected={item.route === route}
                width={100 / NAVIGATION.items.length}
            />
        )
    }

    render() {
        return (
            <Flex
                className='NavigationBar'
                width='100%'
            >
                <Flex
                    className='nav-container'
                    width='100vw'
                >
                    {NAVIGATION.items.map(this.renderNavItem)}
                </Flex>
            </Flex>
        );
    }
}

const mapStateToProps = (state) => {
    return {};
};

const mapDispatchToProps = (dispatch) => {
    return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(NavigationBar);
