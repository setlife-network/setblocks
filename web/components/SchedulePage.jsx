import React from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { cloneDeep } from 'lodash';

import BlockList from './BlockList';
import CommitBlock from './CommitBlock';
import Flex from './Flex';
import ScheduleHeader from './ScheduleHeader';
import SideBar from './SideBar';
import Text from './Text';

import { DEFAULT_SETBLOCKS, TEAM_NAME_ID_MAP } from '../constants';

import {
    fetchCurrentTeamMemberById,
    setEditModeSchedule,
    setEnableSubmit,
    setSelectedDay,
    updateUnsavedSetblocks
} from '../reducers/environment';
import Loading from './Loading';

import theme from '../styles/theme'

class SchedulePage extends React.Component {
    state = {
        daysOfWeek: [],
    }
    
    componentDidMount() {
        const { history, match } = this.props;
        const teamMemberName = match.params.teamMemberName || 'Oscar Lafarga'
        const teamMemberId = match.params.teamMemberId || TEAM_NAME_ID_MAP[teamMemberName]; // Workaround for id until to have a login
        this.props.fetchCurrentTeamMemberById({ id: teamMemberId })
        this.getDaysOfWeek()
        if (!match.params.teamMemberId) {
            const today = moment().toDate();
            // If the match.params don't have a teamMemberId are u seeing your schedule
            history.push('/schedule/' + today.getDay() + '/' + teamMemberName);
            this.props.setEditModeSchedule(true);
            // /schedule - SchedulePage, have today's day selected by default
            this.props.setSelectedDay(today)
        } else {
            // Edit mode in only available in your own schedule
            this.props.setEditModeSchedule(false);
        }

    }

    getDaysOfWeek = () => {
        const startOfWeek = moment().startOf('isoWeek');
        const endOfWeek = moment().endOf('isoWeek');

        const days = [];
        let day = startOfWeek;

        while (day <= endOfWeek) {
            days.push(day.toDate());
            day = day.clone().add(1, 'd');
        }

        this.setState({
            daysOfWeek: days,
        })
        this.props.setSelectedDay(days[0])
    }

    componentDidUpdate (prevProps) {
        const { currentTeamMember, editModeSchedule, selectedDay, history, currentWeeklySetblocks } = this.props;
        // This is to make a different array for editing purpose, completed with empty set blocks
        // This only take effect if change the currentTeamMember
        if (currentTeamMember && currentTeamMember.id === 'error') {
            history.replace('/team'); // If the teamMemberId is invalid should go to /team
        }
        if ((editModeSchedule && currentTeamMember !== prevProps.currentTeamMember) || selectedDay !== prevProps.selectedDay) {
            this.makeSetBlocksForEdit(currentWeeklySetblocks);
            this.props.setEnableSubmit(false)
        }
    }

    makeSetBlocksForEdit = (currentWeeklySetblocks) => {
        const { selectedDay } = this.props;
        const setBlocksByDate = _.groupBy(currentWeeklySetblocks, 'date')
        const day = moment(selectedDay).format('YYYY-MM-DD')
        setBlocksByDate[day] = this.completeWithEmptySetBlocks(setBlocksByDate[day])
        this.props.updateUnsavedSetblocks(setBlocksByDate)
    }

    completeWithEmptySetBlocks(setBlocks) {
        const defaultSetBlocks = cloneDeep(DEFAULT_SETBLOCKS);
        let replacedBlocks = 0;
        setBlocks = _.orderBy(setBlocks, ['blockTime'], ['asc']); // Use Lodash to sort array by 'name'
        setBlocks = _.uniqBy(setBlocks, 'blockTime'); // Use Lodash to delete repeated blockTimes
        if (setBlocks && setBlocks.length > 0) { // To avoid iterate if the array is empty, return default
            defaultSetBlocks.forEach((setBlock, index, theArray) => {
                if (setBlocks && replacedBlocks < setBlocks.length && setBlock.blockTime === setBlocks[replacedBlocks].blockTime) {
                    theArray[index] = setBlocks[replacedBlocks];
                    replacedBlocks++;
                }
            });
        }
        return defaultSetBlocks
    }

    renderMainContent() {
        const { match, currentTeamMember, editModeSchedule, enableSubmit } = this.props

        return (
            <Flex
                center
                column
            >
                <Text
                    weight='900'
                    align='center'
                    mb='0px'
                    style={{ borderBottom: `1px solid ${theme.colors.accent}` }}
                >
                    {match.params.teamMemberId ? currentTeamMember.name + '\'s' : 'Your'}
                    {' Schedule'}
                </Text>
                <BlockList />
                {editModeSchedule && (<CommitBlock enableSubmit={enableSubmit} />)}
            </Flex>
        )
    }

    render() {
        const { selectedDay, loading } = this.props;
        const { daysOfWeek } = this.state;
        return (
            <Flex
                row
                flexDirection='horizontal'
                bg='primary'
                width='100%'
                className='SchedulePage'
            >
                <Flex row>
                    <SideBar
                        days={daysOfWeek}
                        selectedDay={selectedDay}
                    />
                </Flex>
                <Flex
                    column
                    width='100%'
                >
                    <Flex
                        bg={theme.colors.primary}
                        style={{ borderBottom: `2px ${theme.colors.grey} solid` }}
                    >
                        <ScheduleHeader selectedDay={selectedDay} />
                    </Flex>
                    <Loading />
                    {!loading && this.renderMainContent()}
                </Flex>
            </Flex>
        );
    }
}

const mapStateToProps = ({ environment }) => {
    return {
        ...environment,
        loading: environment.pendingNetworkCalls > 0,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        fetchCurrentTeamMemberById: (params) => dispatch(fetchCurrentTeamMemberById(params)),
        setSelectedDay: (selectedDay) => dispatch(setSelectedDay(selectedDay)),
        setEditModeSchedule: (editMode) => dispatch(setEditModeSchedule(editMode)),
        setEnableSubmit: (enableSubmit) => dispatch(setEnableSubmit(enableSubmit)),
        updateUnsavedSetblocks: (unsavedSetBlocks) => dispatch(updateUnsavedSetblocks(unsavedSetBlocks))
    };
};
export default connect(mapStateToProps, mapDispatchToProps)(SchedulePage);
