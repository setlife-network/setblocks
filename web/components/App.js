import React from 'react';
import { compose } from 'redux'
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { Elements, StripeProvider } from 'react-stripe-elements';

import { STRIPE_PUBLIC_KEY } from '../constants'

import Flex from './Flex'
import NavigationBar from './NavigationBar';

import routes from '../routes'

import { checkAuthentication } from '../reducers/auth'

class App extends React.Component {
    componentDidMount() {
        console.log('componentDidMount')
        // Run initialization functions here
        this.props.checkAuthentication()
        .then(loggedIn => {
            // if (loggedIn) this.props.history.push('/team')
        })
    }

    render() {
        const { location } = this.props;

        return (
            <>
                <StripeProvider apiKey={STRIPE_PUBLIC_KEY}>
                    <Flex height='92vh'>
                        <Elements>
                            {routes}
                        </Elements>
                    </Flex>
                </StripeProvider>

                <Flex
                    height='8vh'
                    bg='primary'
                >
                    {this.props.loggedInUser.id != null &&
                        <NavigationBar path={location.pathname} />
                    }
                </Flex>
            </>
        );
    }
}

const mapStateToProps = ({ auth }) => {
    return {
        ...auth
    };
};

const mapDispatchToProps = {
    checkAuthentication
};

export default compose(
    withRouter,
    connect(mapStateToProps, mapDispatchToProps)
)(App)