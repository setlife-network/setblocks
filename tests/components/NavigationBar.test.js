import React from 'react';
import { shallow } from 'enzyme';
import { NavigationBar } from '../../web/components/NavigationBar';

import NavLinkItem from '../../web/components/NavLinkItem';
import Text from '../../web/components/Text';

let renderedComponent;

let mock = () => jest.fn();

describe('NavigationBar Component', () => {
    beforeEach(() => {

        renderedComponent = shallow(
            <NavigationBar
            />
        );
    });

    it('should render correctly', () => {
        expect(renderedComponent).toMatchSnapshot();
    })

});