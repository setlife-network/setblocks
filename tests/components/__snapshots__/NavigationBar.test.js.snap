// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar Component should render correctly 1`] = `
<Flex
  className="NavigationBar"
  width="100%"
>
  <Flex
    className="nav-container"
    width="100vw"
  >
    <_default
      icon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "aria-hidden": [Function],
              "children": [Function],
              "fill": "none",
              "focusable": "false",
              "height": [Function],
              "role": [Function],
              "stroke": "currentColor",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "viewBox": "0 0 24 24",
              "width": [Function],
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": false,
            "rules": Array [
              "
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
",
            ],
          },
          "displayName": "Users",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "svg",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      key="Team"
      route="/team"
      selected={false}
      text="Team"
      width={50}
    />
    <_default
      icon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [
            Object {
              "aria-hidden": [Function],
              "children": [Function],
              "fill": "none",
              "focusable": "false",
              "height": [Function],
              "role": [Function],
              "stroke": "currentColor",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "viewBox": "0 0 24 24",
              "width": [Function],
            },
          ],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "rules": Array [
              "
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
",
            ],
          },
          "displayName": "Calendar",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "svg",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      key="Schedule"
      route="/schedule"
      selected={false}
      text="Schedule"
      width={50}
    />
  </Flex>
</Flex>
`;
